### PR TITLE
Add dataset wijnen 2024

### PR DESCRIPTION
**Pull request by EmilyWes**
A draft pull request to discuss what should and should not be in a pull request for a new dataset. 
General questions:
- Should the enrich.py script be ran before the pull request? (here = yes)
- Should the dataset be added to datasets.toml? (here = not yet)
- Should the attribution directly be changed? (here = not yet)

Note:
- The current gather.py and _ids.csv file used a slightly different enrich.py version that includes support for title/abstract. Some issues (with PR's following soon) are currently open for these changes, some still to be discussed. Before merging I will ensure the _ids file is created with an updated enrich file.

### CODE REVIEW

**Comment by J535D165** [comment]
Thanks for the PR.

A PR should contain the following to make the pipeline run:

- a metadata entry in `datasets.toml`
- a `compose.py` file that outputs a (not enriched) `_ids.csv` file

Optional: 

- The (not enriched) `_ids.csv` file

We will enrich the `_ids.csv` files simultaneously. This should, hopefully, prevent differences in our collection strategy, as OpenAlex frequently updates its data and/or algorithms.

**Comment by EmilyWes** [comment]
Thanks for the answers, running the enrich together makes a lot of sense indeed.
I updated the files.
Shall I also solve the other issues/improvements in the new branch?

File: datasets/Wijnen_2024/compose.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+df.drop_duplicates(subset=["doi","title","year"], inplace=True, ignore_index=True)
+
+# save results to file
+#df.to_csv(f"{key}_raw.csv", index=False)
+
+df_new = df[["doi", "title", "year", "label_included", "label_abstract_included"]].copy()
+df_new["openalex_id"] = None
+
+df_new.to_csv(f"{key}_ids.csv", index=False)
```
Comment (J535D165) [inline datasets/Wijnen_2024/compose.py:37]:
```suggestion
df["openalex_id"] = None

df[["doi", "label_included", "label_abstract_included"]].to_csv(f"{key}_ids.csv", index=False)
```

I know we do this in other files as well, but I think we can simplify to this (my bad)

File: datasets/Wijnen_2024/compose.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+key = "Wijnen_2024"
+
+# Data comes from: https://zenodo.org/records/13957522
+search = ASReviewData.from_file("https://zenodo.org/records/13957522/files/Initial%20search%20(outreach%20included).ris").df
+search["label_included"] = 0
+search["label_abstract_included"] = 0
+
+ti_ab_exclude = ASReviewData.from_file("https://zenodo.org/records/13957522/files/Excluded%20first%20round.ris").df
+ti_ab = search[~search[['primary_title']].apply(lambda x: np.in1d(x,ti_ab_exclude).all(),axis=1)].reset_index(drop=True)
```
Comment (J535D165) [inline datasets/Wijnen_2024/compose.py:14]: Is this an anti-join? I think it's easier to read with pandas `~isin`.

**Review by J535D165 — CHANGES_REQUESTED** [review]
Thanks for the update. Couple of small questions/remarks.

**Comment by J535D165** [comment]
> Thanks for the answers, running the enrich together makes a lot of sense indeed. I updated the files. Shall I also solve the other issues/improvements in the new branch?

Yes, a new branch for the other suggestions such that this remains a single feature branch with a proper title.

File: datasets/Wijnen_2024/compose.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ft = ASReviewData.from_file("https://zenodo.org/records/13957522/files/Included%20papers.ris").df
+ft["label_included"] = 1
+ft["label_abstract_included"] = 1
+
+# Merge and clean up files and drop duplicates (based on ID_SET)
+df = pd.concat([ft,ti_ab,search], ignore_index=True)
+
+df["doi"] = "https://doi.org/" + df["doi"].str.extract(r"(10.\S+)")
+df["year"] = df["publication_year"].str.extract(r"(\d+)")
+df["title"] = df["primary_title"]
```
[the remark this answers is not in the document]
Comment (EmilyWes) [reply datasets/Wijnen_2024/compose.py:27]: Ah, that's the reason, okay.

File: datasets/Wijnen_2024/compose.py (hunk 1 above)
Comment (EmilyWes) [reply datasets/Wijnen_2024/compose.py:37]: Okay, I did add openalex_id to line 36 :)

Comment (J535D165) [reply datasets/Wijnen_2024/compose.py:37]: Oops.

File: datasets/Wijnen_2024/compose.py (hunk 2 above)
Comment (EmilyWes) [reply datasets/Wijnen_2024/compose.py:14]: Yes, indeed looks better now.

**Review by J535D165 — APPROVED** [review]

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
Thanks for the PR and the changes after the review. I am looking forward to your upcoming PRs and datasets.

**Pull request merged**